Orange/evaluation/testing.py

```diff
@@ -268,6 +269,70 @@ def get_augmented_data(self, model_names, include_attrs=True, include_prediction
         predictions.name = data.name
         return predictions
 
+    def fit(self, train_data, test_data=None):
+        test_data = test_data or train_data
+
+        self.prepare_arrays(test_data)
+        n_callbacks = len(self.learners) * len(self.indices)
+        if self.store_models:
+            self.models = np.tile(None, (len(self.indices), len(self.learners)))
+
+        ptr = 0
+        for idx, (train_indices, test_indices) in enumerate(self.indices):
+            train_fold = train_data[train_indices]
+            test_fold = test_data[test_indices]
+
+            if self.preprocessor is not None:
+                train_fold = self.preprocessor(train_fold)
+
+            for k, learner in enumerate(self.learners):
+                model = self.train_if_succ(k, learner, train_fold)
+                self.call_callback((len(self.learners) * idx + k) / n_callbacks)
+                if not model:
+                    continue
+
+                if self.store_models:
+                    self.models[idx][k] = model
+
+                result_slice = slice(ptr, ptr + len(test_fold))
+
+                if train_data.domain.has_discrete_class:
+                    values, probs = model(test_fold, model.ValueProbs)
+                    self.predicted[k][result_slice] = values
+                    self.probabilities[k][result_slice, :] = probs
+                elif train_data.domain.has_continuous_class:
+                    values = model(test_fold, model.Value)
+                    self.predicted[k][result_slice] = values
+                else:
+                    raise Results.DataError("Unknown target type")
+            ptr += len(test_fold)
+
+        self.call_callback(1)
+
+    def prepare_arrays(self, data):
```

Comment:
Could you also add some docstrings to these new base methods? We are now slowly working towards having more docstrings. :smiley: